canonical/specs-v1.canonical.com
# abstract spec preview

## Done
- Added abstract spec preview on click of a spec card title.
- Added new endpoint `/spec/<string:document_id>` that returns the document's html, metadata and respective document url.

## QA
- As we don't have access to docs, you need to login using the app credentials - `credentials.json` (this file can be shared with you).
- Start the project - `dotrun`.
- Login to your Google account.
- Click a spec card title.
- Ensure the abstract preview is displayed.
- Click the close button and ensure the preview is closed.
- Click another card title and ensure the preview is updated.
- Ensure the preview can be focused on with the `Tab` key.
- Ensure that the user is redirected to the respective document on click of the 'Open in Google Docs' button.
